### DISCUSSION TITLE
Unable to reconstruct using 'Generic_Cartesian_Grappa_Complex.xml' ?

### DISCUSSION THREAD

**Discussion by cmn7** [gadgetron/gadgetron#1262] [Q&A]
Hello, 

I am trying to use the Generic_Cartesian_Grappa_Complex.xml reconstruction file with docker-desktop and pulling from ghcr.io/gadgetron/gadgetron/gadgetron_ubuntu_rt_cuda:latest (docker run --net=host -it --rm --gpus=all -v $(pwd):/opt/data ghcr.io/gadgetron/gadgetron/gadgetron_ubuntu_rt_cuda:latest /bin/bash). 

After running siemens_to_ismrmrd successfully to get data and noise .h5 files, I cannot run the *Complex.xml file. It seems to fail during either the BucketToBufferGadget or when GenericReconCartesianReferencePrepGadget starts. Running the default.xml and Generic_Cartesian_Grappa_EPI.xml files works, but we want to save the complex reconstruction data. 

I appreciate any guidance on this, thank you in advance! 

Caitlin

**Comment by hansenms** [comment]
You probably need to add a lot more information in terms of logs and error messages for us to take a look. One quick thing though, I assume you are using the gadgetron_ismrmrmd_client, it may not be able to properly receive the complex images. If you are running this locally, you don't need to do any of that stuff. If you have your converted data file in ismrmrd format, let's say it is `data.h5`, then you can:

```
ismrmrd_hdf5_to_stream -i data.h5 --use-stdout | docker run -i --rm --gpus=all ghcr.io/gadgetron/gadgetron/gadgetron_ubuntu_rt_cuda:latest --from_stream -c Generic_Cartesian_Grappa_Complex.xml | ismrmrd_stream_to_hdf5 --use-stdout -o images.h5
```

You don't have to start a server or use the gadgetron client. More on running the Gagetron in stream mode here: https://gadgetron.readthedocs.io/en/latest/using.html#stream-mode

**Reply by cmn7** [reply]
Thank you so much for the feedback. 

We were incorrectly using the *Complex recon file as we have an EPI sequence. Now we would like to run a version of the Generic_Cartesian_Grappa_EPI.xml recon file, but without the 'ComplexToFloatGadget' so we can save the complex values rather than float values. I created a new file called 'mre_epi.xml' and copied it to the local gadgetron path (/opt/conda/envs/gadgetron/share/gadgetron/config/). 

 After using siemens_to_ismrmrd to convert data and noise, we tried to reconstruct with both mre_epi.xml and Generic_Cartesian_Grappa_EPI.xml using processed data by parameter maps IsmrmrdParameterMap_Siemens.xml (default), and " "_EPI.xml. 

**mre_epi.xml** with the default parameter map runs without errors and outputs imaginary/real components, but most of the output is zeros. 

**Generic_Cartesian_Grappa_EPI.xml** with the default parameter map runs without errors but does not capture imaginary and real components. 

This makes sense, as the only difference between the two is the removal of the ComplexToFloat Gadget. 

However, since most of the data output with mre_epi.xml is zeros, we think there is a incompatibility with the IsmrmrdParameterMap_Siemens_EPI.xsl and the two recon files. Both terminate with errors - 

mre_epi.xml error: 
terminate called after throwing an instance of 'boost::wrapexcept<boost::system::system_error>'
what():  read: End of file [asio.misc:2]
Error caught: send_ismrmrd_acq failed ...
terminate called recursively
Aborted

Generic_Cartesian_Grappa_EPI.xml error: 
Error caught: send_ismrmrd_acq failed ... terminate called after throwing an instance of 'boost::wrapexcept<boost::system::system_error>'
 what():  read: Connection reset by peer [system:104]

Thank you for any feedback or suggestions in advance. 

**edit: I was unable to get stream mode to work without a recurring pipe error, so the above was all done with the gadgetron_ismrmrd_client

**Reply by cmn7** [reply]
Hi, just to follow up on this - do you have any other advice for the errors we are getting? 

Thanks, 
Caitlin

**Reply by hansenms** [reply]
Did you also remove the float to short gadget? Otherwise, you may get some rounding errors.
